### ISSUE TITLE
Table Sort Header not read out well enough by a screen reader

### ISSUE THREAD

**Issue by ArakTaiRoth** [GovAlta/ui-components#2876]
# Info

Reported by Bing

Our Table Sort Header used inside our Table component is read out by screen readers as a button with the header text. There is no mention that it sorts the column, or what direction the column is sorted in.

## Acceptance Criteria:

1. The table sort header component should be read out as a sortable column, and what direction it's currently sorted in if it's currently sorted. If you click to sort, it should then read out what direction the column is currently sorted in.

**Comment by Spark450** [comment]
@ArakTaiRoth Reviewed [this W3C reference implementation](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/#:~:text=Open%20In-,CodePen,-Students%20currently%20enrolled) with our accessibility consultant. They verified that the implementation works well. I would recommend we follow this implementation when working on this issue.
